sqldb/invoices.go

```diff
@@ -0,0 +1,1554 @@
+package sqldb
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+
+	"github.com/lightningnetwork/lnd/channeldb/models"
+	"github.com/lightningnetwork/lnd/clock"
+	invpkg "github.com/lightningnetwork/lnd/invoices"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/record"
+	"github.com/lightningnetwork/lnd/sqldb/sqlc"
+)
+
+// InvoiceQueries is an interface that defines the set of operations that can be
+// executed against the invoice database.
+type InvoiceQueries interface { //nolint:interfacebloat
+	InsertInvoice(ctx context.Context, arg sqlc.InsertInvoiceParams) (int64,
+		error)
+
+	InsertInvoiceFeature(ctx context.Context,
+		arg sqlc.InsertInvoiceFeatureParams) error
+
+	InsertInvoiceHTLC(ctx context.Context,
+		arg sqlc.InsertInvoiceHTLCParams) (int64, error)
+
+	InsertInvoiceHTLCCustomRecord(ctx context.Context,
+		arg sqlc.InsertInvoiceHTLCCustomRecordParams) error
```

Comment:
In `tapd` at times we make type aliases for stuff like this as otherwise the generated types of `InsertInvoiceHTLCCustomRecordParams` can get pretty long.